[MISC] Move seqan3::views::single_pass_input to seqan3/utility/views/

simonsasse opened the pull request:
Resolves: https://github.com/seqan/product_backlog/issues/360 

There was no patch in the issue description so I did not apply any patch.

SGSSGene reviewed (approved):
:+1:

SGSSGene reviewed (changes requested):
You need to move "test/snippet/range/views/single_pass_input.cpp" to test/snippet/utility...

simonsasse commented:
Thanks for the review. I rebased, moved the snippet as you demanded. I also moved the imports starting with seqan3/test back to the position. I separated them from the other seqan3 imports, because I thought it would correct. 👍

simonsasse commented:
I also need to move the seqan3/test/unit/range/views/single_pass_input_test.cpp to seqan3/test/unit/utility/views/single_pass_input_test.cpp.
I'll reassign then !

SGSSGene reviewed (approved):
:+1:

marehr commented on include/seqan3/utility/views/single_pass_input.hpp:
```diff
@@ -0,0 +1,366 @@
+
+#pragma once
+
+#include <seqan3/range/views/detail.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
```
```suggestion
#include <seqan3/std/concepts>
#include <seqan3/std/iterator>
#include <seqan3/std/ranges>

#include <seqan3/range/views/detail.hpp>
```

marehr commented on test/snippet/range/views/take_until.cpp:
```diff
@@ -2,9 +2,9 @@
 #include <seqan3/core/debug_stream.hpp>                 // for debug_stream
-#include <seqan3/range/views/single_pass_input.hpp>     // for views::single_pass_input
 #include <seqan3/range/views/take_until.hpp>            // for views::take_until*
 #include <seqan3/utility/char_operations/predicate.hpp> // for is_char
+#include <seqan3/utility/views/single_pass_input.hpp>     // for views::single_pass_input
```
```suggestion
#include <seqan3/utility/views/single_pass_input.hpp>  // for views::single_pass_input
```

marehr replied:
```suggestion
#include <seqan3/utility/char_operations/predicate.hpp> // for is_char
#include <seqan3/utility/views/single_pass_input.hpp>   // for views::single_pass_input
```

marehr commented:
rebase needed :)
